src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -930,6 +956,55 @@ private void RemoveExtension(IExtension ext)
                 logSource.MessageLogged -= LogMessage;
         }
 
+
+        private void HandleStorageExtensionsOnWorkspaceOpened(WorkspaceModel workspace)
+        {
+            if (!extensionManager.StorageAccessesExtensions.Any())
+                return;
+
+            foreach (var extension in extensionManager.StorageAccessesExtensions)
+            {
+                var ext = extension as IExtension;
+                RaiseIExtensionStorageAccessWorkspaceOpened(workspace, extension, ext.UniqueId);
+            }
+        }
+
+        private void HandleStorageExtensionsOnWorkspaceSaveing(WorkspaceModel workspace, SaveContext saveContext)
+        {
+            if (!extensionManager.StorageAccessesExtensions.Any())
```

Comment:
this check doesn't seem very useful - it's small potatoes since the number of extensions should be small, but we might have to iterate until the last entry in the extensions array for no reason, nothing will happen in the for loop if the StorageAccessExtensions collection is empty.